caffe2/CMakeLists.txt

```diff
@@ -519,37 +518,24 @@ endif()
 # ==========================================================
 
 
-# Compile exposed libraries.
-add_library(caffe2 ${Caffe2_CPU_SRCS})
-
-
-option(TORCH_STATIC "Build libtorch.a rather than libtorch.so" OFF)
-
-
-# This is required for older versions of CMake, which don't allow
-# specifying add_library() without a list of source files
-set(DUMMY_EMPTY_FILE ${CMAKE_BINARY_DIR}/empty.cpp)
-
-if (MSVC)
-  set(DUMMY_FILE_CONTENT "__declspec(dllexport) int ignore_this_library_placeholder(){return 0\\;}")
-else()
-  set(DUMMY_FILE_CONTENT "")
+# Instead of separate .so libraries, GPU sources are now conditionally
+# compiled into the main torch.so library.
+if(USE_CUDA)
+  list(APPEND Caffe2_CPU_SRCS ${Caffe2_GPU_SRCS})
```

Comment:
we should probably rename CPU_SRCS -> SRCS for clarity

Reply:
Agreed.  I plan to take care of the cosmetic stuff in a follow-up to minimize the risk of this PR.